### PR TITLE
Drone Phone!

### PR DESCRIPTION
**Pull request by RemieRichards**
:cl:
add: Added a new PDA button/app: Drone Phone!
/:cl:
## Changes:
- Certain people (Janitors, Engineers, Atmos Techs, Roboticists, HoPs, RDs and Captains) to alert drones.
- `alert_drones()` is now a wrapper around `_alert_drones()`, to keep faction checking but also allowing the proc (now `_alert_drones()`) to be global

DronePhone

### CODE REVIEW

**Comment by Ultimate-Chimera** [comment]
:+1:

**Comment by GunHog** [comment]
RemRem adds the fun! I wish the AI had a proper PDA interface so it could use things like this too.

File: code/modules/mob/living/simple_animal/friendly/drone/say.dm [hunk 1]
```diff
@@ -14,21 +14,34 @@
...
+/mob/living/simple_animal/drone/proc/alert_drones(msg, dead_can_hear = 0)
+	_alert_drones(msg, dead_can_hear, faction)
+
+
 /mob/living/simple_animal/drone/proc/drone_chat(msg)
 	var/rendered = "<i>Drone Chat: \
 		<span class='name'>[name]</span>: \
 		<span class='message'>[say_quote(msg, get_spans())]</span></i>"
-	alert_drones(rendered, 1)
+	alert_drones(rendered, 1, src)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/friendly/drone/say.dm]: This doesn't do anything?

**Comment by Alek2ander** [comment]
Won't responding to a non-drone alert break law 1?

**Comment by Bawhoppen** [comment]
All the drones hear is an electronic ping message, they don't know what causes it whether it was a being or an automatic process.

**Comment by Jordie0608** [comment]
Except the ping explicitly mentions being non-drone.

Even so, this is cool but pretty contrary to the drone policy of non-involvement.

File: code/modules/mob/living/simple_animal/friendly/drone/say.dm [hunk 2]
```diff
@@ -14,16 +14,29 @@
 /mob/living/simple_animal/drone/get_spans()
 	return ..() | SPAN_ROBOT
 
-/mob/living/simple_animal/drone/proc/alert_drones(msg, dead_can_hear = 0)
+
+
+//Base proc for anything to call
+/proc/_alert_drones(msg, dead_can_hear = 0, list/factions)
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/friendly/drone/say.dm]: This is an unconventional way to name a proc, why not something like `alert_drones_output` or `alert_drones_message`

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 3]
```diff
@@ -464,6 +473,7 @@ var/global/list/obj/item/device/pda/PDAs = list()
 							else
 								M.close()
 
+
```
Comment (lordpidey) [inline code/game/objects/items/devices/PDA/PDA.dm]: Truly the most important line in this entire PR.

**Comment by optimumtact** [comment]
@KorPhaeron @lzimann @Shapsy  can you comment on this from an admin perspective? I've already asked some of the admin team about this and the drone policy and feelings are that if we include it it needs to also come along with a quick drone policy update to account for it.

**Comment by Saegrimr** [comment]
Drones can already listen in to binary for reports of damage, carry station bounced radios to listen to common (helps with the boredom) and also respond to reports of shit being broken. I believe the AI also has a way to directly ping drones but I haven't played AI in way too long so i'm not sure about that one.

It's not like a specific drone ping is all that bad, unless theres a lot of people mashing that button for shits and giggles to spam drones.

**Comment by Bawhoppen** [comment]
the drone laws don't actually mean shit, the whole point as drone is to not interfere with the round, whether this violates some arbitrary laws that were just added as an IC way to tell drones what to do doesn't matter that much

**Comment by KorPhaeron** [comment]
Bawhoppen is right

**Comment by Shapsy** [comment]
I'm not really a big fan of encouraging people to just rely on drones to do all the engineering work instead of yelling at their fellow engineers to do it/do it themselves. Why put yourself at risk of injury by patching up that gaping hole in the bar when you can just have a disposable spaceproof drone do it for you?

**Comment by lzimann** [comment]
we should add an option to blow up drones instead of calling them

**Pull request merged**